CMakeLists.txt

```diff
@@ -684,6 +680,19 @@ if(NOT IOS AND NOT XROS)
   include(cmake/OpenCVDetectPython.cmake)
 endif()
 
+# Use same flags for native AArch64 and RISC-V compilation as for cross-compile (Linux)
+if((CV_GCC OR CV_CLANG) AND UNIX AND NOT APPLE AND NOT CMAKE_CROSSCOMPILING AND NOT CMAKE_TOOLCHAIN_FILE)
+  if(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64") # Maybe use AARCH64 variable?
+    include(${CMAKE_CURRENT_LIST_DIR}/platforms/linux/flags-aarch64.cmake)
```

Comment:
There is already something for platforms: https://github.com/opencv/opencv/tree/4.x/cmake/platforms
We could add similar for processors.

Reply:
Perhaps we can move this specific block to `Linux.cmake` if it has all used variables being set at the time of importing (e.g. `CMAKE_CROSSCOMPILING`). I'll try it.

Reply:
Done